File: 002.py
# Problem 2
# Sum of even fibonacci numbers below 4 million.

def sol(z):
    ans = 0
    x,y = 1,2
    
    while x<z:
        if x%2==0:
            ans+=x
        temp = x+y
        x=y
        y=temp

    return ans

print(sol(4000000))

ans = 4613732

File: 002.cpp
// Problem 2
// Sum of even fibonacci numbers below 4 million.

#include <iostream>

int sol(int limit) {
    int sum = 0;
    int prev = 1, current = 2;

    while (current < limit) {
        if (current % 2 == 0) {
            sum += current;
        }
        int next = prev + current;
        prev = current;
        current = next;
    }

    return sum;
}

int main() {
    int outputNumber = sol(4000000);
    std::cout << "Solution: " << outputNumber << std::endl;

    return 0;
}

// ans = 4613732